eng/release/DiagnosticsReleaseTool/Core/Release.cs

```diff
@@ -239,21 +240,30 @@ private async Task<int> LayoutFilesAsync(CancellationToken ct)
                             (FileMapping fileMap, FileMetadata fileMetadata) = layoutResultArray[i];
                             string srcPath = fileMap.LocalSourcePath;
                             string dstPath = fileMap.RelativeOutputPath;
-                            if (relativePublishPathsUsed.Contains(dstPath))
+                            if (relativePublishPathsToHash.ContainsKey(dstPath))
                             {
-                                _logger.LogError("File {srcPath} is getting published to relative path {dstPath} which is already in use.", srcPath, dstPath);
-                                return -1;
+                                if (!string.IsNullOrEmpty(fileMetadata.Sha512) &&
+                                    !string.IsNullOrEmpty(relativePublishPathsToHash[dstPath]) &&
+                                    relativePublishPathsToHash[dstPath] == fileMetadata.Sha512)
+                                {
+                                    _logger.LogInformation("[{buildFilePath}, {worker}, {layoutInd}: {srcPath} -> {dstPath}, {fileMetadata}] File already published to {dstPath} with same hash: {hash}. This is being allowed.", file.FullName, worker.GetType().FullName, i, srcPath, dstPath, fileMetadata, dstPath, fileMetadata.Sha512);
```

Comment:
While this allows the build to succeed, is this what the release team will expect? What's going to happen is that we take a nupkg that was published as to a nuget feed and to a blob feed and only claim that it was published as a NuGetAsset. Do they need to know that it is a BlobAsset as well given that's how it really was published as part of build?

Reply:
AFIK, the release team doesn't care about the format of our publish feed. The only ask I heard was to keep everything in the single container they assigned to us.

Reply:
As a side note, I did look through the AzureBlobPublisher and we will publish the file twice, and verify the contents twice; which is fine. I didn't want to duplicate this logic again, so I left it as-is.